asynctest/case.py

```diff
@@ -340,6 +340,26 @@ class FunctionTestCase(TestCase, unittest.FunctionTestCase):
     """
 
 
+class ClockedTestCase(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.loop.time = lambda: self.time
+        self.time = 0
+
+    def advance(self, seconds):
+        assert seconds >= 0
+        self._drain_loop()
+        self.time += seconds
+        self._drain_loop()
+
+    def _drain_loop(self):
+        while self.loop._ready or any(map(
+                lambda handle: handle._when <= self.time,
+                self.loop._scheduled)):
+            self.loop._run_once()
```

Comment:
I'm not sure, what I've tended to find is what you normally end up writing is something like:
- set up conditions
- assertions
- simulate wait
- assertions

So I guess it depends on how you're writing the tests.
If you're using run_until_complete then it'll be easier as is, otherwise the coroutine will be better.
I am happy to go with whichever is more idiomatic for the lib.